src/genie/libs/parser/iosxr/show_snmp.py

```diff
@@ -0,0 +1,547 @@
+
+        # 68954 Number of altered variables
+        p7 = re.compile(r"^(?P<altered_variables>\d+)\s+Number\s+of\s+altered\s+variables$")
+
+        # 22102104 Get-request PDUs
+        p8 = re.compile(r"^(?P<get_request_pdus>\d+)\s+Get-request\s+PDUs$")
+
+        # 972411 Get-next PDUs
+        p9 = re.compile(r"^(?P<get_next_pdus>\d+)\s+Get-next\s+PDUs$")
+
+        # 3878 Set-request PDUs
+        p10 = re.compile(r"^(?P<set_request_pdus>\d+)\s+Set-request\s+PDUs$")
+
+        # 42531179 SNMP packets output
+        p11 = re.compile(r"^(?P<output_pkts>\d+)\s+SNMP\s+packets\s+output$")
+
+        # 0 Too big errors (Maximum packet size 1500)
+        p12 = re.compile(r"^(?P<big_errors>\d+)\s+Too\s+big\s+errors\s+\(Maximum\s+packet\s+size\s+(?P<max_pkt_size>\d+)\)$")
+
+        # 160671 No such name errors
+        p13 = re.compile(r"^(?P<name_errors>\d+)\s+No\s+such\s+name\s+errors$")
+
+        # 0 Bad values errors
+        p14 = re.compile(r"^(?P<value_errors>\d+)\s+Bad\s+values\s+errors$")
+
+        # 0 General errors
+        p15 = re.compile(r"^(?P<general_errors>\d+)\s+General\s+errors$")
+
+        # 34854547 Response PDUs
+        p16 = re.compile(r"^(?P<response_pdus>\d+)\s+Response\s+PDUs$")
+
+        # 7676632 Trap PDUs
+        p17 = re.compile(r"^(?P<trap_pdus>\d+)\s+Trap\s+PDUs$")
+
+        # SNMP logging: Enabled
+        p18 = re.compile(r"^SNMP\s+logging:\s+(?P<logging_status>Enabled|Disabled)$")
+
+        # Logging to Notification host: 10.0.0.1, udp-port: 162
+        p19 = re.compile(r"^Logging\sto\sNotification\shost:\s(?P<logging_host>[a-fA-F\d.:]+),\sudp-port:\s(?P<udp_port>\d+)$")
+
+        # Trap Host Statistics
+        p20 = re.compile(r"^(?P<notification_type>(Trap|Inform))\sHost\sStatistics$")
+
+        # Number of pkts in Trap Queue: 0
+        p21 = re.compile(r"^Number\sof\spkts\sin\sTrap\sQueue:\s(?P<pkts_in_trap_queue>\d+)$")
+
+        # Maximum length of Trap Queue: 100
+        p22 = re.compile(r"^Maximum\slength\sof\sTrap\sQueue:\s(?P<max_len_trap_queue>\d+)$")
+
+        # Number of pkts sent: 0
+        p23 = re.compile(r"^Number\sof\spkts\ssent:\s(?P<pkts_sent>\d+)$")
+
+        # Number of pkts dropped: 0
+        p24 = re.compile(r"^Number\sof\spkts\sdropped:\s(?P<pkts_dropped>\d+)$")
+
+        # Number of Informs sent: 0
+        p25 = re.compile(r"^Number\sof\sInforms\ssent:\s(?P<informs_sent>\d+)$")
+
+        # Number of Informs retries: 0
+        p26 = re.compile(r"^Number\sof\sInforms\sretries:\s(?P<informs_retries>\d+)$")
+
+        # Number of Informs pending: 0
+        p27 = re.compile(r"^Number\sof\sInforms\spending:\s(?P<informs_pending>\d+)")
+
+        # Number of Informs dropped: 0 
+        p28 = re.compile(r"^Number\sof\sInforms\sdropped:\s(?P<informs_dropped>\d+)$")
+
+        snmp_dict = {}
+
+        for line in output.splitlines():
+            if line:
+                line = line.strip()
+            else:
+                continue
+
+            m = p0.match(line)
```

Comment:
Above each match, please comment an example line that will be matched, exactly like you've done with the compiled regex

Reply:
changes made as requested